src/main/perl/Blockdevices.pm

```diff
@@ -105,6 +107,270 @@ sub should_create_ks
 
 =pod
 
+=head2 is_correct_device
+
+Returns true if this is the device that corresponds with the device 
+described in the profile.
+
+The method can log an error, as it is more of a sanity check then a test.
+
+=cut
+
+sub is_correct_device
+{
+    my $self = shift;
+    # Legacy behaviour is no checking; always assuming correct device.
+    $this_app->info ("is_correct_device method not defined. Returning true for legacy behaviour.");
+    return 1;
+}
+
+# Returns size in byte (assumes devpath exists).
+# Is used by size
+sub _size_in_byte
+{
+    my $self = shift;
+    my $size = CAF::Process->new([GET_SIZE_BYTES, $self->devpath], log => $this_app)->output();
+    chomp($size);
+    return $size;
+}
+
+=pod
+
+=head2 size
+
+Returns size in MiB if the device exists in the system. 
+Returns undef if the device doesn't exist.
+
+=cut
+
+sub size
+{
+    my $self = shift;
+    my $size;
+    if ($self->devexists) {
+        my $bytes = $self->_size_in_byte();
+        $size = $bytes / (1024 * 1024);
+        $this_app->verbose("Device $self->{devname}, has size $size MiB ($bytes byte)");
+    } else {
+        $this_app->verbose("No size for device $self->{devname}, devpath ",
+                           $self->devpath, " doesn't exist");
+    }
+    return $size;
+}
+
+=pod 
+
+=head2 correct_size_interval
+
+Compute the smallest interval for the expected C<size>
+given the conditions (one or more of):
+
+=over
+
+=item fraction : double
+
+The difference between the found and the expected size is at most 
+C<fraction> of the expected size. C<fraction> is a double 
+(e.g. 1 percent is 0.01).
+
+=item diff : long
+
+The difference between the found and the expected size (in MiB) is at most 
+C<diff> MiB.
+
+=back
+
+=cut
+
+sub correct_size_interval
+{
+    my $self= shift;
+
+    my @conds = sort keys %{$self->{correct}->{size}};
+    $this_app->verbose("Going to use ", scalar @conds, " conditions: ", join(", ", @conds));    
+
+    my ($min, $max);
+
+    my $update_min_max = sub {
+        my ($nmin, $nmax) = @_;
```

Comment:
What do `$min`, `$max`, `$nmin` and `$nmax` refer to?
I'm a little confused :confused: